test/modifiers/batch.test.js

```diff
@@ -0,0 +1,284 @@
+    runBasicStreamTests(data, objData, function (stream, objectMode, done) {
+      var actual = [];
+
+      stream
+        .pipe(batch(batchOptions))
+        .on('data', function (chunk) {
+          actual.push(chunk);
+        })
+        .on('error', done)
+        .on('end', function () {
+          if (objectMode) {
+            expect(actual).to.deep.equal(objExpected);
+          } else {
+            expect(actual).to.deep.equal(expected);
+          }
+
+          done();
+        });
+    });
+
+    it('writes all chunks as an array', function (done) {
+      var input = getReadableStream([1, 2, 3], {
+        objectMode: true
+      });
+
+      var chunksRead = false;
+
+      input
+        .pipe(batch(batchOptions))
+        .on('data', function (chunk) {
+          chunksRead = true;
+          expect(chunk).to.be.an('array');
+        })
+        .on('error', done)
+        .on('end', function () {
+          expect(chunksRead).to.equal(true);
+          done();
+        });
+    });
+
+    it('successfully ends if there is no data written', function (done) {
+      var input = getReadableStream([], {
+        objectMode: true
+      });
+
+      var chunksRead = false;
+
+      input
+        .pipe(batch(batchOptions))
+        .on('data', function (chunk) {
+          chunksRead = true;
+          expect(chunk).to.be.an('array');
+        })
+        .on('error', done)
+        .on('end', function () {
+          expect(chunksRead).to.equal(false);
+          done();
+        });
+    });
+
+  }
+
+  describe('when "time" is defined in options', function () {
+    var expected = [[buff('item1')], [buff('item2'), buff('item3'), buff('item4')]];
+    var objExpected = [
+      objData.slice(0, 1),
+      objData.slice(1)
+    ];
+
+    basicTests({ time: 5 }, expected, objExpected);
+
+    it('combines data written during an interval into one write', function (done) {
+      var actual = [];
+      var input = through.obj();
+
+      input
+        .pipe(batch({ time: 5 }))
+        .on('data', function (chunk) {
+          actual.push(chunk);
+        })
+        .on('error', done)
+        .on('end', function () {
+          expect(actual).to.deep.equal([[1], [2, 3], [4, 5]]);
+
+          done();
+        });
+
+      input.write(1);
+
+      setTimeout(function () {
+        input.write(2);
+        input.write(3);
+      }, 1);
+
+      setTimeout(function () {
+        input.write(4);
+        input.write(5);
+
+        input.end();
+      }, 6);
+    });
+  });
+
+  describe('when "count" is defined in options', function () {
+    var expected = [[buff('item1'), buff('item2')], [buff('item3'), buff('item4')]];
+    var tempObjData = [].concat(objData);
+    var objExpected = [];
+
+    while (tempObjData.length) {
+      objExpected.push(tempObjData.splice(0, 2));
+    }
+
+    basicTests({ count: 2 }, expected, objExpected);
+
+    it('emits chunks of the defined size', function (done) {
```

Comment:
I see this test and the one before it expect a result that is precisely the correct length. If I had a stream with 9 items and a count of 2, would that emit 5 times with the appropriate data?